### ISSUE TITLE
Wrong download is resumed when restarted

### ISSUE THREAD

**Issue by sharkwouter** [sharkwouter/minigalaxy#507]
The last game which was downloading should be downloaded when Minigalaxy is restarted. Instead if downloads the game which was last added to the download queue.

**Comment by GB609** [comment]
From what i could tell, minigalaxy currently requeues all downloads for all games.
The queueing order more or less depends on the construction order of GameTile(List), which ultimately is the iteration order of Library.games, see `Library.__add_gametile`.
With the current config, up to 4 game files are downloaded in parallel.

We'd first have to separate the library listing from the download restarting.
What we could do is determine download progress per game and requeue them in another order after the library listing has finished.

Possible order:
1. Percentage high -> low - this is not necessarily the fastest to complete, a game might have 50 gb, 1% of that is still more than e.g. 500 mb from another game.
2. total remaining bytes low -> high
3. as defined in config.current_downloads - after i reworked the logic that updates the list, the order will remain and not be changed by queueing other games

**Comment by sharkwouter** [comment]
I honestly don't know if this is still a problem since we implemented the download manager and downloading multiple games at once.

**Comment by GB609** [comment]
Yeah, the order itself maybe wouldn't be as much of a problem. This'll only start becoming an issue again if there are more then 4 game files pending

**Comment by sharkwouter** [comment]
It seems this still exists.

**Comment by GB609** [comment]
i've started working on a fix for the order already. The first part was the addition of `add_ongoing_download` and `remove_ongoing_download` to config to make sure the list retains its order.
The next step is under test, see https://github.com/GB609/minigalaxy/tree/bugfix/507

And the last part i want to change is the logic in LibraryEntry that builds the list of files to download because it does very strange things:
```
self.download_list = []
files_to_download = []
iterate files:
  files_to_download.insertFirst(file) #just why

self.download_list.extend(files_to_download)

download_finish_func:
  install(self.download_list[-1]) # pick last file, because they are added in inverse order
```
1. Why use a local, and extends download_list when download list is emptied before.
2. what is the point and use of reversing the list in the first place?

**Comment by sharkwouter** [comment]
I think we could also reverse the list when we first load it. I'll do a test with that.

**Comment by sharkwouter** [comment]
I was thinking something like this: https://github.com/sharkwouter/minigalaxy/commit/0e209994c0a928f41636fb8254a81606b5da7d42

**Comment by GB609** [comment]
Double post, but easier to track than my frequent edits:
Once the download UI is there and i've enhanced download manager to differentiate between 'abort' and 'pause', this might become less of an issue.

**Comment by sharkwouter** [comment]
Makes sense!

**Comment by sharkwouter** [comment]
Did we fix this one?

**Comment by GB609** [comment]
I think we can close this one. With the download ui in place, the order can be seen and manipulated by the user.

**Comment by sharkwouter** [comment]
That makes sense, done
